dotnet/src/Connectors/Connectors.Memory.Postgres/PostgresServiceCollectionExtensions.cs

```diff
@@ -13,112 +13,87 @@ namespace Microsoft.SemanticKernel;
-            });
-
-        return services;
-    }
-
-    /// <summary>
-    /// Register a Postgres <see cref="IVectorStore"/> with the specified service ID and where <see cref="IPostgresVectorStoreDbClient"/> is retrieved from the dependency injection container.
-    /// </summary>
-    /// <param name="services">The <see cref="IServiceCollection"/> to register the <see cref="IVectorStore"/> on.</param>
-    /// <param name="options">Optional options to further configure the <see cref="IVectorStore"/>.</param>
-    /// <param name="serviceId">An optional service id to use as the service key.</param>
-    /// <returns>The service collection.</returns>
-    public static IServiceCollection AddPostgresVectorStore(this IServiceCollection services, PostgresVectorStoreOptions? options = default, string? serviceId = default)
-    {
-        // Since we are not constructing the client, add the IVectorStore as transient, since we
-        // cannot make assumptions about how client is being managed.
-        services.AddKeyedTransient<IVectorStore>(
+        services.AddKeyedSingleton<IVectorStore>(
             serviceId,
             (sp, obj) =>
             {
-                var client = sp.GetRequiredService<IPostgresVectorStoreDbClient>();
+                var dataSource = sp.GetRequiredKeyedService<NpgsqlDataSource>(npgsqlServiceId);
                 var selectedOptions = options ?? sp.GetService<PostgresVectorStoreOptions>();
 
                 return new PostgresVectorStore(
-                    client,
+                    dataSource,
                     selectedOptions);
             });
 
         return services;
     }
 
     /// <summary>
-    /// Register a Postgres <see cref="IVectorStore"/> with the specified service ID and where <see cref="IPostgresVectorStoreDbClient"/> is constructed using the provided parameters.
+    /// Register a Postgres <see cref="IVectorStore"/> with the specified service ID and where an NpgsqlDataSource is passed in.
     /// </summary>
     /// <param name="services">The <see cref="IServiceCollection"/> to register the <see cref="IVectorStore"/> on.</param>
-    /// <param name="connectionString">Postgres database connection string.</param>
+    /// <param name="dataSource">The data source to use.</param>
     /// <param name="options">Optional options to further configure the <see cref="IVectorStore"/>.</param>
     /// <param name="serviceId">An optional service id to use as the service key.</param>
     /// <returns>The service collection.</returns>
-    public static IServiceCollection AddPostgresVectorStore(this IServiceCollection services, string connectionString, PostgresVectorStoreOptions? options = default, string? serviceId = default)
+    public static IServiceCollection AddPostgresVectorStore(this IServiceCollection services, NpgsqlDataSource dataSource, PostgresVectorStoreOptions? options = default, string? serviceId = default)
```

Comment:
I'm not convinced that this overload is that useful. Where someone builds their DI container, they will not typically be able to create a `NpgsqlDataSource` in a way that will get it destroyed in the right timeframe.  E.g. if they use a using clause it'll just get destroyed immediately before the app really starts.  If they register it with DI to allow it to handle it's lifecycle, they will use a different overload.

FYI: We typically follow a philosophy of avoiding adding extra overloads or helpers unless we have a good justification, simply because it means more maintenance and more chances of breaking changes in future.  So less is more 😄